inductiva/molecules/scenarios/water_box/water_box.py

```diff
@@ -0,0 +1,145 @@
+from inductiva.scenarios import Scenario
+from inductiva.utils.files import remove_files_with_tag
+
+SCENARIO_TEMPLATE_DIR = os.path.join(TEMPLATES_PATH, "water_box")
+GROMACS_TEMPLATE_INPUT_DIR = "gromacs"
+
+
+class WaterBox(Scenario):
+    """Water box scenario."""
+
+    def __init__(
+        self,
+        temperature: float = 300,
+        box_size: float = 3.0,
+    ):
+        """The scenario involves simulating a box filled with water molecules.
+        The simulation consists of three main steps: energy minimization, water 
+        molecule position decorrelation, and molecular dynamics simulation. 
+        Currently, this scenario exclusively supports the GROMACS simulator.
+        By default, for a given box size the water molecules are always 
+        initially positioned in the same locations following a uniform
+        distribution. 
+        To introduce randomness in the initial positions of the water molecules, 
+        a decorrelation step is performed. This step involves running a short 
+        simulation so that their position changes according to a normal 
+        distribution.
+        Args:
+            temperature: The temperature of the simulation in Kelvin.
+            box_size: The size of the box in nm.
+        """
+        self.template_dir = os.path.join(SCENARIO_TEMPLATE_DIR,
+                                         GROMACS_TEMPLATE_INPUT_DIR)
+        self.temperature = temperature
+        self.box_size = box_size
+
+    def simulate(
+            self,
+            simulator: Simulator = GROMACS(),
+            output_dir: Optional[Path] = None,
+            simulation_time: float = 10,  # ns
+            integrator: Literal["md", "sd", "bd"] = "md",
+            nsteps_minim: int = 5000):
+        """Simulate the water box scenario.
+
+        Args:
+            output_dir: The output directory to save the simulation results.
+            simulation_time: The simulation time in ns.
+            integrator: The integrator to use for the simulation. Options:
+                - "md" (Molecular Dynamics): Accurate leap-frog algorithm for 
+                integrating Newton's equations of motion.
+                - "sd" (Steepest Descent): Stochastic dynamics integrator with
+                leap-frog scheme.
+                - "bd" (Brownian Dynamics): Euler integrator for Brownian or 
+                position Langevin dynamics. 
+                
+            For more details on the integrators, refer to the GROMACS 
+            documentation at 
+            https://manual.gromacs.org/current/user-guide/mdp-options.html.
+
+            nsteps_minim: Number of steps for energy minimization.
+        """
+        self.nsteps = int(
+            simulation_time * 1e6 / 2
+        )  # convert to fs and divide by the time step of the simulation (2 fs)
+        self.integrator = integrator
+        self.nsteps_minim = nsteps_minim
+        replace_params_in_template(
```

Comment:
This part should go in the gen_config, no?

Reply:
since the commands go directly in the super().simulate(commands=commands) I need to read it before the call. Because the gen_config is called inside super.simulate().

Reply:
Seems good! Maybe later we can tackle this if it repeats in other scenarios.